plugins/kotlin/uast/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/FirKotlinUastResolveProviderService.kt

```diff
@@ -155,50 +156,39 @@ interface FirKotlinUastResolveProviderService : BaseKotlinUastResolveProviderSer
     override fun resolveBitwiseOperators(ktBinaryExpression: KtBinaryExpression): UastBinaryOperator {
         val other = UastBinaryOperator.OTHER
         analyseForUast(ktBinaryExpression) {
-            val resolvedCall = ktBinaryExpression.resolveCall()?.targetFunction?.getSingleCandidateSymbolOrNull() ?: return other
-            val operatorName = resolvedCall.callableIdIfNonLocal?.callableName?.asString() ?: return other
+            val resolvedCall = ktBinaryExpression.resolveCall()?.singleCallOrNull<KtFunctionCall<*>>() ?: return other
+            val operatorName = resolvedCall.symbol.callableIdIfNonLocal?.callableName?.asString() ?: return other
             return KotlinUBinaryExpression.BITWISE_OPERATORS[operatorName] ?: other
         }
     }
 
     override fun resolveCall(ktElement: KtElement): PsiMethod? {
-        when (ktElement) {
-            is KtCallElement -> {
-                analyseForUast(ktElement) {
-                    return ktElement.resolveCall()?.let { toPsiMethod(it) }
-                }
-            }
-            is KtBinaryExpression -> {
-                analyseForUast(ktElement) {
-                    return ktElement.resolveCall()?.let { toPsiMethod(it) }
-                }
-            }
-            is KtUnaryExpression -> {
-                analyseForUast(ktElement) {
-                    return ktElement.resolveCall()?.let { toPsiMethod(it) }
-                }
-            }
-            else ->
-                return null
+        return analyseForUast(ktElement) {
+            ktElement.resolveCall()?.singleCallOrNull<KtFunctionCall<*>>()?.symbol?.let { toPsiMethod(it) }
```

Comment:
`singleCallOrNull<KtFunctionCall<*>>()` is repeated. How about introducing variants of utils per type parameter? e.g., `singleFunctionCallOrNull`, `singleVariableAccessCallOrNull`, `singleConstructorCallOrNull`, etc.?

Reply:
Sounds good. Let me add these helpers.